hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestBulkLoadReplication.java

```diff
@@ -0,0 +1,304 @@
+ * It also defines a preBulkLoad coprocessor that is added to all test table regions on each of the
+ * clusters, in order to count amount of times bulk load actually gets invoked. This is to certify
+ * we are not entered in the infinite loop condition addressed by HBASE-22380.
+ */
+@Category({ ReplicationTests.class, MediumTests.class})
+public class TestBulkLoadReplication extends TestReplicationBase {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestBulkLoadReplication.class);
+
+  protected static final Logger LOG =
+    LoggerFactory.getLogger(TestBulkLoadReplication.class);
+
+  private static final String PEER1_CLUSTER_ID = "peer1";
+  private static final String PEER2_CLUSTER_ID = "peer2";
+  private static final String PEER3_CLUSTER_ID = "peer3";
+
+  private static final String PEER_ID1 = "1";
+  private static final String PEER_ID3 = "3";
+
+  private static final AtomicInteger BULK_LOADS_COUNT = new AtomicInteger(0);
+  private static CountDownLatch BULK_LOAD_LATCH;
+
+  private static final HBaseTestingUtility UTIL3 = new HBaseTestingUtility();
+  private static final Configuration CONF3 = UTIL3.getConfiguration();
+  private static Table htable3;
+
+  @Rule
+  public TestName name = new TestName();
+
+  @ClassRule
+  public static TemporaryFolder testFolder = new TemporaryFolder();
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    setupBulkLoadConfigsForCluster(CONF1, PEER1_CLUSTER_ID);
+    setupBulkLoadConfigsForCluster(CONF2, PEER2_CLUSTER_ID);
+    setupBulkLoadConfigsForCluster(CONF3, PEER3_CLUSTER_ID);
+    setupConfig(UTIL3, "/3");
+    TestReplicationBase.setUpBeforeClass();
+    startThirdCluster();
+  }
+
+  private static void startThirdCluster() throws Exception {
+    LOG.info("Setup Zk to same one from UTIL1 and UTIL2");
+    UTIL3.setZkCluster(UTIL1.getZkCluster());
+    UTIL3.startMiniCluster(NUM_SLAVES1);
+
+    TableDescriptor table = TableDescriptorBuilder.newBuilder(tableName)
+      .setColumnFamily(ColumnFamilyDescriptorBuilder.newBuilder(famName).setMaxVersions(100)
+        .setScope(HConstants.REPLICATION_SCOPE_GLOBAL).build())
+      .setColumnFamily(ColumnFamilyDescriptorBuilder.of(noRepfamName)).build();
+
+    Connection connection3 = ConnectionFactory.createConnection(CONF3);
+    try (Admin admin3 = connection3.getAdmin()) {
+      admin3.createTable(table, HBaseTestingUtility.KEYS_FOR_HBA_CREATE_TABLE);
+    }
+    UTIL3.waitUntilAllRegionsAssigned(tableName);
+    htable3 = connection3.getTable(tableName);
+  }
+
+  @Before
+  @Override
+  public void setUpBase() throws Exception {
+    super.setUpBase();
+    ReplicationPeerConfig peer1Config = getPeerConfigForCluster(UTIL1);
+    ReplicationPeerConfig peer2Config = getPeerConfigForCluster(UTIL2);
+    ReplicationPeerConfig peer3Config = getPeerConfigForCluster(UTIL3);
+    //adds cluster1 as a remote peer on cluster2
```

Comment:
Is this what you intended? 2 replicates to 1 and 3, and 3 goes back to 2?

Not 1 to 2 to 3 to 1?